File: myjsapp/src/json.js
dateObj = [null,null,2021,2,15,null,4,null,6];

function getRecMessagesStringJSON(reqid, b64, subnum) {
    var reqdata1 = [[
        reqid,
        null,
        null,
        null,
        null,
        b64,
        dateObj
    ]];

    return reqdata1;
}

function getAckMessagesStringJSON(newguid, guids, n64, subnum) { 
    var resjson =  [[
        newguid,
        null,
        null,
        null,
        null,
        n64,
        dateObj
    ],guids,null,[]

    ];

    return resjson;
}

function getSendMessagesStringJSON(bugle, respguid, respauth, sendproto64) { 

    var resjson =  [bugle,
    [
        respguid,
        19,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        sendproto64
    ],
    [
        respguid,
        null,
        null,
        null,
        null,
        respauth,
        dateObj
    ],
    null,
    86400000000,
    null,
    null,
    null,
    []
    ];

    return resjson;
}

function GetRefreshTokenJSON(guid, oldtachyon, bugle, unixtimestamp, signedcode) {
    var json = [
        [
        guid,null,null,null,null,
        oldtachyon,
        dateObj
        ],
        bugle,
        unixtimestamp,
        signedcode,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
            null,
        [null,null,null,null,null,null,null,null,[]
        ]
        ,
        null
        ,
        null,2];

        return json;
}


module.exports = {
    getRecMessagesStringJSON,
    getSendMessagesStringJSON,
    getAckMessagesStringJSON,
    GetRefreshTokenJSON
}

File: mygoapp/dhmessagesapi/messages/json.go
package messages

import (
	"encoding/json"
)

func getRecMessagesStringJSON(reqid string, b64 string) string {
	dateObj := []interface{}{nil, nil, 2021, 2, 15, nil, 4, nil, 6}

	reqdata1 := [][]interface{}{
		{
			reqid,
			nil,
			nil,
			nil,
			nil,
			b64,
			dateObj,
		},
	}

	jsonData, _ := json.Marshal(reqdata1)

	return string(jsonData)
}

func getAckMessagesStringJSON(newguid string, guids []string, n64 string) (string, error) {
	dateObj := []interface{}{nil, nil, 2021, 2, 15, nil, 4, nil, 6}

	resjson := []interface{}{
		[]interface{}{
			newguid,
			nil,
			nil,
			nil,
			nil,
			n64,
			dateObj,
		},
		guids,
		nil,
		[]interface{}{},
	}

	jsonData, err := json.Marshal(resjson)
	if err != nil {
		return "", err
	}

	return string(jsonData), nil
}

func GetRefreshTokenJSON(guid, oldtachyon string, BugleId int32, BugleNum string, Bugle string, unixtimestamp int64, signedcode interface{}) []interface{} {

	bugleobj := []interface{}{BugleId, BugleNum, Bugle}

	dateObj := []interface{}{nil, nil, 2021, 2, 15, nil, 4, nil, 6}

	json := []interface{}{
		[]interface{}{
			guid,
			nil,
			nil,
			nil,
			nil,
			oldtachyon,
			dateObj,
		},
		bugleobj,
		unixtimestamp,
		signedcode,
		nil,
		nil,
		nil,
		nil,
		nil,
		nil,
		nil,
		nil,
		[]interface{}{
			nil,
			nil,
			nil,
			nil,
			nil,
			nil,
			nil,
			nil,
			[]interface{}{},
		},
		nil,
		nil,
		2,
	}

	return json
}

func getSendMessagesStringJSON(BugleId int32, BugleNum string, Bugle string, respguid, respauth, sendproto64 string) (string, error) {

	bugleobj := []interface{}{BugleId, BugleNum, Bugle}
	dateObj := []interface{}{nil, nil, 2021, 2, 15, nil, 4, nil, 6}
	resjson := []interface{}{
		bugleobj,
		[]interface{}{
			respguid,
			19,
			nil,
			nil,
			nil,
			nil,
			nil,
			nil,
			nil,
			nil,
			nil,
			sendproto64,
		},
		[]interface{}{
			respguid,
			nil,
			nil,
			nil,
			nil,
			respauth,
			dateObj,
		},
		nil,
		86400000000,
		nil,
		nil,
		nil,
		[]interface{}{},
	}

	resjsonBytes, err := json.Marshal(resjson)
	if err != nil {
		return "", err
	}

	return string(resjsonBytes), nil
}
